plugins/wmap.rb

```diff
@@ -1316,7 +1318,13 @@ def add_web_site(url)
           ssl = true
         end
 
-        site = framework.db.report_web_site(wait: true, host: uri.host, port: uri.port, vhost: vhost, ssl: ssl, workspace: framework.db.workspace)
+        site = begin
+          framework.db.report_web_site(wait: true, host: uri.host, port: uri.port, vhost: vhost, ssl: ssl, workspace: framework.db.workspace)
+        rescue SocketError => e
```

Comment:
Upstream's logic behind making a layer7 protocol failure (DNS resolution) present as a layer 4 condition aside, this ought to still work correctly with the native resolver (https://github.com/rapid7/rex-socket/pull/43) code as the exception that throws emulates this one from the standard library.